logstash-core-event-java/src/main/java/org/logstash/Event.java

```diff
@@ -338,22 +348,33 @@ private Timestamp initTimestamp(Object o) {
             logger.warn("Error parsing " + TIMESTAMP + " string value=" + o.toString());
         }
 
-        tag(TIMESTAMP_FAILURE_TAG);
-        this.data.put(TIMESTAMP_FAILURE_FIELD, o);
-
-        return Timestamp.now();
+        return null;
     }
 
     public void tag(String tag) {
```

Comment:
as per our discussion we keep current logic. I added comments and switched to use Arrays.asList() for more concision.